src/librustc_parse/parser/module.rs

```diff
@@ -1,15 +1,30 @@
-use crate::{new_sub_parser_from_file, Directory, DirectoryOwnership};
+use crate::new_sub_parser_from_file;
 
 use rustc_ast::ast::{self, Attribute, Ident, Mod};
-use rustc_ast::attr;
-use rustc_ast::token;
+use rustc_ast::{attr, token};
 use rustc_errors::{struct_span_err, PResult};
 use rustc_session::parse::ParseSess;
 use rustc_span::source_map::{FileName, Span};
 use rustc_span::symbol::sym;
 
 use std::path::{self, Path, PathBuf};
 
+#[derive(Clone)]
+pub struct Directory {
```

Comment:
I don't think it should be in `rustc_parse` at all.
Are there any other places besides modules using it?

Reply:
Done in a next commit.